src/gobeventproducer/producer.py

```diff
@@ -132,100 +184,56 @@ def _init_gob_db_session(self):
-        return last_event.last_event if last_event else -1
-
-    def _set_last_eventid(self, eventid: int):
-        last_event = self._get_last_event()
-
-        if last_event:
-            last_event.last_event = eventid
-        else:
-            last_event = LastSentEvent(
-                catalogue=self.catalogue, collection=self.collection, last_event=eventid)
-            self.db_session.add(last_event)
-
-        self.db_session.commit()
-
-    def _get_events(self, min_eventid: int):
+    def _get_events(self, min_eventid: int, max_eventid: int):
         return self.gob_db_session \
             .query(self.Event) \
             .yield_per(10000) \
             .filter(and_(self.Event.catalogue == self.catalogue,
                          self.Event.entity == self.collection,
-                         self.Event.eventid > min_eventid)) \
+                         self.Event.eventid > min_eventid,
+                         self.Event.eventid <= max_eventid)) \
             .order_by(self.Event.eventid.asc())
 
-    def _add_event(self, event):
+    def _add_event(self, event, connection):
         header = {
             'event_type': event.action,
             'event_id': event.eventid,
             'tid': event.tid,
             'catalog': event.catalogue,
             'collection': event.entity,
         }
-        headers = [(k, _to_bytes(str(v)) if v else b'') for k, v in header.items()]
         data = self.event_builder.build_event(event.tid)
 
-        self.producer.send(
-            KAFKA_TOPIC,
-            key=_to_bytes(header['tid']),
-            value=_to_bytes(json.dumps(data)),
-            headers=headers
-        )
+        msg = {
+            'header': header,
+            'data': data
+        }
+        connection.publish(EVENTS_EXCHANGE, self.routing_key, msg)
 
-    def _flush(self, last_eventid: int):
-        self.producer.flush(timeout=120)
-        self._set_last_eventid(last_eventid)
-        print(f"Flushed Kafka events. Total events: {self.total_cnt}. Last event id: {last_eventid}")
+    def produce(self, min_eventid: int, max_eventid: int):
+        """Produce external events starting from min_eventid (exclusive) until max_eventid (inclusive)."""
+        start_eventid = min_eventid
 
-    def produce(self):
-        last_eventid = self._get_last_eventid()
-        self.logger.info(f"Start producing. Last event was {last_eventid}")
+        with LocalDatabaseConnection(self.catalogue, self.collection) as localdb:
+            last_eventid = localdb.get_last_eventid()
 
-        events = self._get_events(last_eventid)
+            # Ideally we would remove the need for the database. We keep the database in place now to be able to spot
+            # any errors thay may arise when min_eventid does not match the expected last_eventid.
+            if last_eventid != min_eventid:
+                self.logger.warning(f"Min eventid ({min_eventid}) to produce does not match last_eventid "
```

Comment:
Maybe a more descriptive message here in case of no last_eventid == -1?
No last_eventid found, start producing from first event 

`Min eventid (370691) to produce does not match last_eventid (-1) in database. Recovering.
2023-02-27T11:04:13.680611961Z Start producing events > -1 and <= 370691`